plugins/drm/lib/DrmLicenseUtils.php

```diff
@@ -0,0 +1,96 @@
+<?php
+/**
+ * @abstract
+ * @plugin drm
+ * @package plugins.drm
+ * @subpackage api.services
+ */
+
+class DrmLicenseUtils {
+
+    const SYSTEM_NAME = 'OVP';
+    
+    private $secureEntryHelper;
+
+    public function __construct($entry, $referrer)
+    {
+        $this->secureEntryHelper = new KSecureEntryHelper($entry, kCurrentContext::$ks, $referrer, ContextType::PLAY);
+        $this->secureEntryHelper->validateForPlay();
+    }
+
+    /**
+     * @return mixed $policyId
+     */
+    public function getPolicyId()
+    {
+        KalturaLog::debug("Validating access control");
+        $actions = $this->secureEntryHelper->getContextResult()->getActions();
+        $policyId = null;
+        foreach($actions as $action)
+        {
+            if($action instanceof kAccessControlDrmPolicyAction && $action->getPolicyId())
+            {
+                $policyId = $action->getPolicyId();
+                break;
+            }
+        }
+        return $policyId;
+    }
+
+    public static function calculateExpirationDate(DrmPolicy $policy, entry $entry)
+    {
+        $beginDate = time();
+        switch($policy->getLicenseExpirationPolicy())
+        {
+            case DrmLicenseExpirationPolicy::FIXED_DURATION:
+                $expirationDate = $beginDate + dateUtils::DAY*$policy->getDuration();
+                break;
+            case DrmLicenseExpirationPolicy::ENTRY_SCHEDULING_END:
+                $expirationDate = $entry->getEndDate();
+                break;
+        }
+        return $expirationDate;
+    }
+
+    public function getSecureEntryHelper()
+    {
+        return $this->secureEntryHelper;
+    }
+
+    public static function signDataWithKey($dataToSign, $signingKey)
+    {
+        return rawurlencode(base64_encode(sha1($signingKey.$dataToSign,TRUE)));
+    }
+
+	public static function createCustomData($entryId, KalturaFlavorAssetArray $flavorAssets, $signingKey)
+	{
+		$flavorParamIds = "";
+		foreach ($flavorAssets as $flavor)
+		{
+			$flavorParamIds .= $flavor->flavorParamsId.",";
+		}
+
+		$innerData = new stdClass();
```

Comment:
Please use associative array instead of stdClass, it will be encoded the same into JSON.